content/articles/constructors-in-kotlin/index.md

```diff
@@ -0,0 +1,180 @@
+```
+Name = Alex Email = Alex@gmail
+```
+
+Constructors can also be called by another secondary constructor of the same class:
+
+```kotlin
+class myStudent{
+     constructor(name:String,id:Int): this(name,id,"Alex@gmail"){
+        println("Name=${name}")
+         println("Id=${id}")
+    }
+    constructor(name:string, id:Int, mail:string ) 
+      { 
+          println("Name = ${name}") 
+          println("Id" = ${id})
+           println("Email = ${mail}")
+     }
+}    
+     fun main(args:Array<string>){
+     val mystudent = myStudent("Alex",200) 
+    
+}
+```
+The output of the above code is:
+
+```
+Name = Alex 
+Id = 200
+Email = Alex@gmail
+```
+
+### Init blocks
+Init short for initializer. It is a block of code that is executed every time the class is instantiated. To declare one, use the `init` keyword:
+
+```kotlin
+init
+{ 
+    //code to be executed
+}
+```
+
+Here is an example of a primary constructor with an init block:
+
+```kotlin
+class person(name:string,id:Int){ 
+    val pName :String 
+    val pId:Int 
+    init{ 
+        pName= name.capitalize() 
+        pId=id println("Name = ${pName}") 
+        println("Id=${PId}") 
+        } 
+    } fun main(args:Array<Strings>){ 
+        val Person=person("Alex",200) 
+}
+```
+
+When ran this would output:
+
+```
+Name = ALEX 
+Id =200
+```
+
+### The difference between a primary constructor and a secondary constructor
+The main difference between a primary and secondary constructor is that the former resides in the class declaration header, whereas the latter is declared inside the Kotlin class body. Additionally, a secondary constructor may call the primary constructor.
+
+### Best practices for object-oriented programming
+The clean use of classes is required for effective Kotlin. We learned about constructors above but never discussed best practices with them. We can make the following changes to our code to make it more readable while using constructors:
+- Minimize visibility of properties, i.e., private, public, protected. It allows you to control where your class members can be accessed from.
+- Combine property declaration. This is achieved by taking a property and setting it to a constructor parameter.
+- Improve variable naming. This is achieved by using variable naming rules; your variable names should: 
+  - Be based on the variable's purpose so it's clear what the variable is for.
+  - Have no spaces between words. Ex. myClass.
+  - Be greater than a single character.
+  - Not use special prefixes or suffixes (e.g. name_, mName, s_name, and kName), except in the case of backing properties.
+
+### Conclusion
+In this article, we have covered various ways of creating constructors using Kotlin as our language, and when to use the init block. We also learned that the init block is called just after the primary constructor, and the secondary constructor is called using the `this` keyword. Finally, we went through best practices in object-oriented programming. 

```

Comment:
You never explicitly said that the init block is called just after the primary constructor. While you did use a code snippet above to demonstrate it, you shouldn't assume that the reader will pick it up just by looking at the output. Please add a short explanation detailing that after showing the output (the output I'm talking about is on line 161).

Reply:
I believe there is a mistake here: you said:

> The secondary constructor is called using the `this` keyword.

You never mentioned this above and I don't think that is correct. Did you mean the primary constructor is called using the `this` keyword?